tests/components/statistics/test_init.py

```diff
@@ -15,3 +17,93 @@ async def test_unload_entry(hass: HomeAssistant, loaded_entry: MockConfigEntry)
     assert await hass.config_entries.async_unload(loaded_entry.entry_id)
     await hass.async_block_till_done()
     assert loaded_entry.state is ConfigEntryState.NOT_LOADED
+
+
+async def test_device_cleaning(
+    hass: HomeAssistant,
+    device_registry: dr.DeviceRegistry,
+    entity_registry: er.EntityRegistry,
+) -> None:
+    """Test for source entity device for Statistics."""
```

Comment:
Function name and docstring do not make sense. Also, shouldn't these be split up into two-three test cases?

Reply:
Edit: I can see that the first half of this testcase is a replica of the test case defined in the other file. Was that necessary?

Reply:
>  Also, shouldn't these be split up into two-three test cases?

Can you specify which three you suggest?

> Was that necessary?

What do you suggest? Deduplication? IMHO, that isn't needed for 2 occurrences.

Reply:
No I wouldn't spend that time. Many of the test cases start with similar initialization code.
@dougiteixeira I would appreciate a minimal config and a better docstring. After that the addition lgtm!! 🚀

Reply:
The sensor test only checks whether the device id of the source entity was linked to the helper entity.

The init test, on the other hand, cleans up links between devices other than the source entity (the user can rename entities and impact the integration). That's why the initial part is the same, as I need to link a valid device as done in the sensor test and some stale devices.

Reply:
I improved the docstring.

As I need to configure the integration to trigger cleaning, I don't see a way to reduce this.